include/swift/AST/TypeNodes.def

```diff
@@ -131,9 +131,13 @@ ABSTRACT_TYPE(AnyMetatype, Type)
 ALWAYS_CANONICAL_TYPE(Module, Type)
 TYPE(DynamicSelf, Type)
 ABSTRACT_TYPE(Substitutable, Type)
-  ALWAYS_CANONICAL_TYPE(Archetype, SubstitutableType)
+  ABSTRACT_TYPE(Archetype, SubstitutableType)
+    ALWAYS_CANONICAL_TYPE(ContextArchetype, ArchetypeType)
```

Comment:
I hate to make you rename everything but we've called these "primary archetypes" elsewhere

Reply:
No problem. I'll also rename `ChildArchetype` to `NestedArchetype`, which jives better with the terminology used elsewhere.